### PR TITLE
uvma_rvfi_instr_mon.sv: fix a non compliant LRM SystemVerilog code

### PR DESCRIPTION
**Pull request by ZElkacimi**
#https://github.com/openhwgroup/core-v-verif/issues/1268#issue-1235352499
Signed-off-by: Zineb El Kacimi <zineb.el-kacimi@external.thalesgroup.com>

### CODE REVIEW

**Review by MikeOpenHWGroup — COMMENTED** [review]
Hi @ZElkacimi, this fix looks good to me.  Before merging I will ask @silabs-robin to review as he was one of the authors of this agent and it is used extensively at his organization (Silicon Labs).

**Comment by silabs-robin** [comment]
> I will ask @silabs-robin to review

Thanks for considering me regarding this change.
I tried to review it but I didn't fully understand it, so I just have that one question above if either of you could help explain.

File: lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv
Comment (silabs-robin) [reply lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv:208]: (This might possibly be one of those cases where I again wish we were using `default_nettype none`. But I'm not sure how that works on the oop side of systemverilog.)

File: lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv
Comment (silabs-robin) [inline lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv:213]: Once we are at it, maybe we should add a named constant instead of the magic number 3?

File: lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv
Comment (MikeOpenHWGroup) [reply lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv:208]: > (I don't have a way of easily running this myself.)

This is a library component, so if you clone @ZElkacimi's fork/branch and run any test of the E40X/S that uses the RVFI instruction monitor (virtually every test!), you should be able to see it in action.

File: lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv
Comment (silabs-robin) [reply lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv:213]: I am not surprised 😄 Sorry about that. I shall do my best to motivate my future self to utmost diligence.

Comment (silabs-robin) [reply lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv:213]: Hi. I have my comments above. I did not understand how the code would work, and it seems Xcelium also could not compile it.

Comment (silabs-robin) [reply lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv:213]: Is it possible for you to add a breakpoint in the code when running simulation? What happens if you add a breakpoint on this line?: https://github.com/openhwgroup/core-v-verif/blob/611850728c0f6fa4a8e5da0a60a3bf4d7270c0c1/lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv#L208
To me, it seems like it is defined in a different scope and should not be visible from that scope, and it seems Xcelium agrees with me. @MikeOpenHWGroup , could you try this on e.g. dsim or replicate on one of the simulators we use?

Comment (ZElkacimi) [reply lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv:213]: Trying to set the BP, I realized that I did not create the rvfi agent correctly, that's why the compilator didn't detect these errors before. Now, I have same errors you mentioned above;  "Identifier 'get_csr_retirement_data' has not been declared yet ..." 

I think this agent was functional, one day, since it's in lib/ , maybe you can debug, if if you have any project where you used it for the first time. Or what do you think?

Comment (silabs-robin) [reply lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv:213]: The agent should be functional. I think it compiled on the commit right before the one in this PR.
The fix itself is also a good idea (if the existing code is truly not LRM compliant), so I think what needs to be done is to work on the code a bit to get the fix working.

**Comment by ZElkacimi** [comment]
Hello @silabs-robin
Sorry, It was me who proposed a fix that not make sense. Please try this one, and let me know if it works better.

File: lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv
Comment (silabs-robin) [inline lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv:233]: Could we name it `dcsr_ret_data` so that usage of the variable makes it clear which CSR is being referenced?

Comment (ZElkacimi) [reply lib/uvm_agents/uvma_rvfi/uvma_rvfi_instr_mon.sv]: > Could we name it `dcsr_ret_data` so that usage of the variable makes it clear which CSR is being referenced?

Done

**Review by silabs-robin — APPROVED** [review]
Approving on the basis of: The fix makes sense to me and hello-world runs without problem.

**Review by MikeOpenHWGroup — APPROVED** [review]

**Comment by MikeOpenHWGroup** [comment]
Thanks all.

**Pull request merged**
